resources/services/ecs/clusters.go

```diff
@@ -1076,34 +1462,46 @@ func fetchEcsClusters(ctx context.Context, meta schema.ClientMeta, parent *schem
 	}
 	return nil
 }
-func resolveEcsClustersDefaultCapacityProviderStrategy(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
-	cluster := resource.Item.(types.Cluster)
+func resolveClustersDefaultCapacityProviderStrategy(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	cluster, ok := resource.Item.(types.Cluster)
+	if !ok {
+		return diag.WrapError(fmt.Errorf("expected to have types.Cluster but got %T", resource.Item))
+	}
 	data, err := json.Marshal(cluster.DefaultCapacityProviderStrategy)
 	if err != nil {
 		return diag.WrapError(err)
 	}
 	return resource.Set(c.Name, data)
 }
-func resolveEcsClustersSettings(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
-	cluster := resource.Item.(types.Cluster)
+func resolveClustersSettings(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	cluster, ok := resource.Item.(types.Cluster)
+	if !ok {
+		return diag.WrapError(fmt.Errorf("expected to have types.Cluster but got %T", resource.Item))
+	}
```

Comment:
I don't think we are checking the type casts anymore. Right @roneli?

Reply:
We can remove this assumption as it usually 100% true, it does avoid a panic, but it should never happen.